### ISSUE TITLE
Failed to boot ISO when using the "ostreecontainer" directive in kickstart

### ISSUE THREAD

**Issue by ggiguash** [osbuild/bootc-image-builder#330]
When booting an ISO image using `virt-install` with a kickstart file containing the [ostreecontainer](https://pykickstart.readthedocs.io/en/latest/kickstart-docs.html#ostreecontainer) directive, the system fails to boot the resulting image.

Here's what we are doing in our build and run procedures.

**Building the base ISO**
```
sudo podman run --rm -i --privileged --security-opt label=type:unconfined_t \
   -v /home/microshift/microshift/_output/test-images/bootc-iso-images/centos9-bootc:/output \
   quay.io/centos-bootc/bootc-image-builder:latest  \
   --type anaconda-iso \
   quay.io/centos-bootc/centos-bootc:stream9
```

**Building a container layer**
This can be as simple as just creating a directory

**Booting a VM**
```
sudo virt-install --autoconsole text --graphics none \
   --name cos9-src-standard-suite2-host1 --vcpus 2 --memory 4096 \
   --disk pool=vm-storage-cos9-src@standard-suite2,size=20 \
   --network network=default,model=virtio --events on_reboot=restart --noreboot \
   --location /home/ggiguash/Projects/microshift/_output/test-images/vm-storage/centos9-bootc.iso,kernel=images/pxeboot/vmlinuz,initrd=images/pxeboot/initrd.img \
   --osinfo detect=on \
   --filesystem=/home/ggiguash/Projects/microshift/_output/test-images/bootc-images,bootc-images,driver.type=virtiofs \
   --memorybacking=source.type=memfd,access.mode=shared \
   --extra-args "fips=0 console=ttyS0,115200n8 inst.notmux inst.ks=file:/kickstart.HMbv1ngK.ks"
   --initrd-inject /tmp/kickstart.HMbv1ngK.ks \
   --wait
```

**Kickstart**
The file is pretty standard, containing the required partitioning, etc. The only `special` thing we're doing is inserting the `ostreecontainer` directive which pulls the image from a file system mounted from the hyperisor. I suspect, the behavior will be the same when pulling the image from registry.

```
...
...
zerombr
clearpart --all --initlabel
part biosboot --fstype=biosboot --size=1
part /boot --fstype=ext4 --asprimary --size=1000
# Uncomment this line to add a SWAP partition of the recommended size
#part swap --fstype=swap --recommended
part pv.01 --grow
volgroup rhel pv.01
logvol / --vgname=rhel --fstype=xfs --size=REPLACE_LVM_SYSROOT_SIZE --name=root

...
...
# Mount the bootc image directory from the hypervisor
%pre-install --log=/dev/console --erroronfail

mkdir -p /mnt/bootc-images
mount -v -t virtiofs bootc-images /mnt/bootc-images

%end

# Configure bootc
ostreecontainer --transport oci --url /mnt/bootc-images/REPLACE_BOOT_COMMIT_REF --no-signature-verification
```

**Comment by ondrejbudai** [comment]
Can you give us more information? I would like to see:
- how you built the ISO
- the kickstart file
- how you booted the ISO

**Comment by ggiguash** [comment]
> Can you give us more information? I would like to see:
> 
> * how you built the ISO
> * the kickstart file
> * how you booted the ISO

@ondrejbudai, I updated the description of the issue with more information.

**Comment by ggiguash** [comment]
I made some progress in troubleshooting this problem. The OS starts to boot, but it cannot mount the root file system. 
Here are the errors I'm getting from the `systemd-fsck.service`
![image](https://github.com/osbuild/bootc-image-builder/assets/85498834/20c4711f-ecc7-4423-8380-c25ecb04c0c5)

**Comment by ondrejbudai** [comment]
Ah, this is probably due to the fact that our installer is still using Fedora which creates a filesystem that RHEL/Centos cannot read. Should be fixed by #338.

**Comment by ggiguash** [comment]
@ondrejbudai , I confirm that the boot of the OS works OK when the base ISO is built using the code from the current PR.
